src/CMakeLists.txt

```diff
@@ -253,40 +254,57 @@ source_group("types" FILES ${types_files})
 source_group("utils" FILES ${utils_files})
 
 #--------------------------------------------------------
-# Lib
-add_library(${src_lib_static_nupiccore_solo} STATIC 
+# Lib core_alg (encoders, SP, TM/TP/BackTM, Anomaly,...) 
+add_library(${src_lib_static_nupiccore_alg} STATIC 
     ${algorithm_files} 
     ${encoders_files} 
-    ${engine_files} 
     ${math_files} 
-    ${ntypes_files} 
-    ${os_files} 
-    ${regions_files} 
     ${types_files} 
     ${utils_files} 
 )
 
 
-target_compile_options( ${src_lib_static_nupiccore_solo} PUBLIC ${INTERNAL_CXX_FLAGS})
-target_compile_definitions(${src_lib_static_nupiccore_solo} PRIVATE ${COMMON_COMPILER_DEFINITIONS})
-target_include_directories(${src_lib_static_nupiccore_solo} PRIVATE 
+target_compile_options( ${src_lib_static_nupiccore_alg} PUBLIC ${INTERNAL_CXX_FLAGS} ${CXX_FLAGS_OPTIMIZED})
+target_compile_definitions(${src_lib_static_nupiccore_alg} PRIVATE ${COMMON_COMPILER_DEFINITIONS})
+target_include_directories(${src_lib_static_nupiccore_alg} PRIVATE 
 		${CORE_LIB_INCLUDES} 
 		${EXTERNAL_INCLUDES}
 		)
+set_target_properties(${src_lib_static_nupiccore_alg} PROPERTIES LINK_FLAGS "${LINKER_FLAGS_OPTIMIZED}")
 
 if(${NUPIC_IWYU})
-  set_target_properties(${src_lib_static_nupiccore_solo}
+  set_target_properties(${src_lib_static_nupiccore_alg}
     PROPERTIES CXX_INCLUDE_WHAT_YOU_USE ${iwyu_path})
 endif()
 
 
+# Lib NetworkAPI (Regions, Engine,...) 
+add_library(${src_lib_static_nupiccore_networkapi} STATIC
+  ${engine_files}
+  ${ntypes_files}
+  ${os_files}
+  ${regions_files}
+)
+
+target_compile_options( ${src_lib_static_nupiccore_networkapi} PUBLIC ${INTERNAL_CXX_FLAGS})
+target_compile_definitions(${src_lib_static_nupiccore_networkapi} PRIVATE ${COMMON_COMPILER_DEFINITIONS})
+target_include_directories(${src_lib_static_nupiccore_networkapi} PRIVATE
+  ${CORE_LIB_INCLUDES}
+  ${EXTERNAL_INCLUDES}
+)
+if(${NUPIC_IWYU})
+  set_target_properties(${src_lib_static_nupiccore_networkapi}
+    PROPERTIES CXX_INCLUDE_WHAT_YOU_USE ${iwyu_path})
+endif()
+
 #--------------------------------------------------------
 #
 # Create the nupic_core as a "combined" static library by merging nupic_core_solo
 # with our external static libraries
 #
 set(src_combined_nupiccore_source_archives
-    ${src_lib_static_nupiccore_solo}
+    ${src_lib_static_nupiccore_alg}
```

Comment:
I see no reason to separate them.

Reply:
more modularity to give you option to use the optimal flags (speed, code stability, checks,..) for each of them.

Reply:
I don't see the need.  But if you do want to split NetworkAPI into its own library then you need to move some files around.
- ntypes/MemParser.cpp/hpp  and MemStream.hpp are not part of NetworkAPI
- utils/ Watcher.cpp/hpp is part of NetworkAPI and not algorithms
- There are some files that belong in both libraries.